lib/datasource/galaxy-collection/index.ts

```diff
@@ -0,0 +1,156 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import * as packageCache from '../../util/cache/package';
+import { Http } from '../../util/http';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export const id = 'galaxy-collection';
+export const defaultRegistryUrls = ['https://galaxy.ansible.com/'];
+export const customRegistrySupport = false;
+
+const http = new Http(id);
+
+interface BaseProjectResult {
+  versions_url: string;
+  deprecated: boolean;
+  latest_version: {
+    version: string;
+  };
+}
+
+interface VersionsProjectResult {
+  results: Versions[];
+}
+
+interface VersionsDetailResult {
+  download_url: string;
+  artifact: {
+    filename: string;
+    size: bigint;
+    sha256: string;
+  };
+  metadata: {
+    homepage: string;
+    tags: Record<string, string>;
+    dependencies: Record<string, string>;
+    repository: string;
+  };
+}
+
+interface Versions {
+  version: string;
+  href: string;
+}
+
+export async function getReleases({
+  lookupName,
+  registryUrl,
+}: GetReleasesConfig): Promise<ReleaseResult | null> {
+  const cacheNamespace = 'datasource-galaxy-collection';
+  const cacheKey = lookupName;
+  const cachedResult = await packageCache.get<ReleaseResult>(
+    cacheNamespace,
+    cacheKey
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const lookUp = lookupName.split('.');
+  const namespace = lookUp[0];
+  const projectName = lookUp[1];
+
+  const baseUrl = `${registryUrl}api/v2/collections/${namespace}/${projectName}/`;
+
+  try {
+    const baseUrlResponse: any = await http.get(baseUrl);
+    if (!baseUrlResponse || !baseUrlResponse.body) {
+      logger.warn(
+        { dependency: lookupName },
+        `Received invalid data from ${baseUrl}`
+      );
+      return null;
+    }
+
+    const baseProject: BaseProjectResult = JSON.parse(baseUrlResponse.body);
+
+    const versionsUrl = `${baseUrl}versions/`;
+    const versionsUrlResponse: any = await http.get(versionsUrl);
```

Comment:
```suggestion
    const versionsUrlResponse = await http.getJson<VersionsProjectResult>(versionsUrl);
```
Use typed json response

Reply:
It should work, maybe you need to debug why the calls, maybe there is a hidden exception you don't expect.

It can be  a non json parseble reponse like `.reply(200, '')` 😉

Reply:
K, found the issue. It has been a copy and paste error :facepalm: